Challenge the skier with the monster

guyonvarch opened the pull request:
Make the monster to really follow the skier, and stop after some time.

This also:

- reduce a bit the collision area of the skier going south, permitting the monster not to catch the skier when going from side to south,
- fix spawning side trees,
- rollback to distance for metrics,
- increase distance from a given amount of pixels to prevent having a smaller score.

Commit:
Make the monster more interesting

Make it follow the player according to his speed, and to be tired and
stop the chase, either after some time, or after the skier makes a jump.

Commit:
Use distance as before in metrics for backward compat

Commit:
Fix spawning side trees

Commit:
Reduce spawning snowboarders

knuton commented on src/config.ts:
```diff
@@ -3,7 +3,7 @@ import * as Vec2 from 'lib/vec2'
   scaling: window.devicePixelRatio,
   originalFrameInterval: 20,
-  pixelsPerMeter: (diagonal: number) => 18 * diagonal / 2000,
+  pixelsPerMeter: (diagonal: number) => diagonal / 250,
```
Explain?

guyonvarch replied:
We divide the diagonal in other places as well, what do you have in mind for this place in particular ?

knuton replied:
I thought it would be worth to explain that the particular weight is chosen to produce results that are roughly sensible in comparison with previous results.

guyonvarch replied:
It’s a currently important point concerning this change, to respect a similar or bigger distance is reached. Though, once the change will be applied, don’t you think it would be in the way to talk about a previous version of the code ?

knuton replied:
If you think it is only of historical interest (personally I am not so sure, as we want to maintain the compatibility and it usually helps to understand the origin of any magic number), it might make sense to capture the context in the commit message instead of the code.

knuton commented on src/lib/random.ts:
```diff
@@ -1,3 +1,7 @@
+export function float({ min, max }: { min: number, max: number }): number {
+  return min + Math.random() * (max - min + 1)
```
This can produce values greater than `max`.

guyonvarch replied:
Exact: https://github.com/dividat/skifree.js/pull/10/commits/ea06c5e823fa25bb421915e668f0ac55d7c2ca8c https://github.com/dividat/skifree.js/pull/10/commits/0960d647e7aa488a71f7911f88bc45f72cfe4859

knuton reviewed (commented):
Looking good in high level code review, practical testing coming soon.

Two small things to improve.

Commit:
Fix off by one Random.float

Commit:
Fix drop rates after Random.float adjustment

knuton commented on src/config.ts:
```diff
@@ -3,7 +3,7 @@ import * as Vec2 from 'lib/vec2'
   scaling: window.devicePixelRatio,
   originalFrameInterval: 20,
-  pixelsPerMeter: (diagonal: number) => 18 * diagonal / 2000,
+  pixelsPerMeter: (diagonal: number) => diagonal / 250,
```
Starting a second thread here: This specific conversion weight produces a very significant jump for me, from an average distance of ~4600 in the old version to closer to ~7400 in this PR.

How do the distance metrics between versions change for you?

Commit:
Augment landing width a bit

It was sometimes a bit narrow, going a bit on the sides just after
landing could make the skier to hit a tree.

Commit:
Increase pixels per meter to compute distance travelled

I played with the following handicaped:

- have to look backward to play,
- with only one eye,
- semi-closed to see blurry,
- through a tunnel reducing the area of vision made by my hand.

With those handicaps, to get the same score with the old version and
the new one, the magic number is 200.

When playing normally, it’s 160.

So, drawing the line in the middle at 180.
